operator/controllers/keptntask/function_utils.go

```diff
@@ -0,0 +1,138 @@
+package keptntask
+
+import (
+	"encoding/json"
+	"fmt"
+	klcv1alpha1 "github.com/keptn-sandbox/lifecycle-controller/operator/api/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"math/rand"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type FunctionExecutionParams struct {
+	ConfigMap        string
+	Parameters       map[string]string
+	SecureParameters string
+	URL              string
+}
+
+func (r *KeptnTaskReconciler) generateFunctionJob(task *klcv1alpha1.KeptnTask, params FunctionExecutionParams) (*batchv1.Job, error) {
+	randomId := rand.Intn(99999-10000) + 10000
+	jobId := fmt.Sprintf("klc-%s-%s-%s-%d", task.Spec.Application, task.Spec.Workload, task.Spec.WorkloadVersion, randomId)
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobId,
+			Namespace: task.Namespace,
+			Annotations: map[string]string{
+				"keptn.sh/app":      task.Spec.Application,
+				"keptn.sh/workload": task.Spec.Workload,
+				"keptn.sh/version":  task.Spec.WorkloadVersion,
+			},
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy: "OnFailure",
+				},
+			},
+		},
+	}
+	err := controllerutil.SetControllerReference(task, job, r.Scheme)
+	if err != nil {
+		r.Log.Error(err, "could not set controller reference:")
+	}
+
+	container := corev1.Container{
+		Name:  "keptn-function-runner",
+		Image: "ghcr.io/keptn-sandbox/functions-runtime:main.202209211413",
+	}
+
+	var envVars []corev1.EnvVar
+
+	if len(params.Parameters) > 0 {
+		jsonParams, err := json.Marshal(params.Parameters)
+		if err != nil {
+			return job, fmt.Errorf("could not marshal parameters")
+		}
+		envVars = append(envVars, corev1.EnvVar{Name: "DATA", Value: string(jsonParams)})
+	}
+
+	if params.SecureParameters != "" {
+		envVars = append(envVars, corev1.EnvVar{
+			Name: "SECURE_DATA",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{Name: params.SecureParameters},
+					Key:                  "SECURE_DATA",
+				},
+			},
+		})
+	}
+
+	if params.ConfigMap != "" {
```

Comment:
Here I would leave a comment saying that the TaskDefinition controller mounts inline functions as CM

Reply:
changed